vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java

```diff
@@ -58,4 +101,88 @@ public void sendData(String data, String topic) throws InterruptedException, Exe
     {
         kafkaTemplate.send(topic, data).get(WAIT_TIMEOUT_IN_MINUTES, TimeUnit.MINUTES);
     }
+
+    /**
+     * Starts the Kafka consumer with the provided configuration to listen the specified topics. The consumer must be
+     * stopped when it's not needed.
+     *
+     * @param topics the comma-separated set of topics to listen
+     */
+    @When("I start consuming messages from Kafka topics `$topics`")
+    public void startKafkaListener(Set<String> topics)
+    {
+        BlockingQueue<String> messageQueue = new LinkedBlockingDeque<>();
+        testContext.put(MESSAGES_KEY, messageQueue);
+        ContainerProperties containerProperties = new ContainerProperties(topics.toArray(new String[0]));
+        containerProperties.setMessageListener(
+                (MessageListener<String, String>) data -> messageQueue.add(data.value()));
+        KafkaMessageListenerContainer<String, String> container = new KafkaMessageListenerContainer<>(consumerFactory,
+                containerProperties);
+        container.start();
+        testContext.put(LISTENER_KEY, container);
+        LOGGER.info("Kafka message listener is started");
+    }
+
+    /**
+     * Waits until the count of the consumed messaged (from the consumer start or after the last draining operation)
+     * matches to the rule or until the timeout is exceeded.
+     *
+     * @param timeout        the maximum time to wait for the messages in ISO-8601 format
+     * @param comparisonRule The rule to compare values
+     *                       (<i>Possible values:<b> less than, less than or equal to, greater than, greater than or
+     *                       equal to,
+     *                       equal to</b></i>)
+     * @param expectedCount  the expected count of the messages to be matched by the rule
+     */
+    @When("I wait with `$timeout` timeout until count of consumed Kafka messages is $comparisonRule `$expectedCount`")
+    public void waitForKafkaMessages(Duration timeout, ComparisonRule comparisonRule, int expectedCount)
+    {
+        Matcher<Integer> countMatcher = comparisonRule.getComparisonRule(expectedCount);
+        Integer result = new DurationBasedWaiter(timeout, Duration.ofSeconds(1)).wait(
```

Comment:
Could you please clarify why the polling duration is a hardcoded value?

Reply:
we poll the data from memory (`testContext`), so it's cheap, 
IMHO, 1 second is a period which could save test time and at the same time it's not so much for humans (like, 1s or 0.5s - it doesn't matter for users)

Reply:
<img src="https://user-images.githubusercontent.com/5081226/100100387-c8cd1e00-2e71-11eb-8e08-a662968e023b.png" height="200"/>